s3.tf

```diff
@@ -1,48 +1,73 @@
 resource "aws_s3_bucket" "web" {
   bucket = var.primary_fqdn
+}
 
-  acl = "public-read"
+resource "aws_s3_bucket_acl" "web_acl" {
+  bucket = aws_s3_bucket.web.id
+  acl    = "public-read"
+}
 
-  website {
-    index_document = var.web_index_doc
-    error_document = var.web_error_doc
+resource "aws_s3_bucket_website_configuration" "web_site_configuration" {
+  bucket = aws_s3_bucket.web.id
 
-    routing_rules = var.routing_rules
+  index_document {
+    suffix = var.web_index_doc
   }
 
+  error_document {
+    key = var.web_error_doc
+  }
+
+  routing_rules = var.routing_rules
+}
+
+resource "aws_s3_bucket_cors_configuration" "web_cors_configuration" {
+  bucket = aws_s3_bucket.web.id
+
   cors_rule {
     allowed_headers = ["*"]
     allowed_methods = ["GET", "HEAD"]
     allowed_origins = var.origins
     expose_headers  = var.cors_expose_headers
     max_age_seconds = var.cors_max_age_seconds
   }
-
-  policy = <<EOF
-{
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Sid": "",
-            "Effect": "Allow",
-            "Principal": {
-                "AWS": "${aws_cloudfront_origin_access_identity.access_id.iam_arn}"
-            },
-            "Action": "s3:GetObject",
-            "Resource": "arn:aws:s3:::${var.primary_fqdn}/*"
-        },
-        {
-            "Sid": "",
-            "Effect": "Allow",
-            "Principal": {
-                "AWS": "${aws_cloudfront_origin_access_identity.access_id.iam_arn}"
-            },
-            "Action": "s3:ListBucket",
-            "Resource": "arn:aws:s3:::${var.primary_fqdn}"
-        }
-    ]
 }
-EOF
 
+resource "aws_s3_bucket_policy" "web_policy" {
+  bucket = aws_s3_bucket.web.id
+  policy = data.aws_iam_policy_document.web_policy_document.json
 }
 
+data "aws_iam_policy_document" "web_policy_document" {
```

Comment:
love

Reply:
Is this meant to be `data` and not a `resource`? I'm not sure how we would expect for this `aws_iam_policy_document` to already exist unless it is being generated by this terraform, and I'm not seeing where that would happen.

Reply:
From what I understand, this isn't creating a resource or pulling one from somewhere else. It simply represents data that can be used where needed.

Reply:
Interesting. It was always my understanding that `data` blocks represented an existing piece of infrastructure defined outside of the terraform code in question. I'll read up on this, thank you!